crates/pathfinder/src/merkle_tree/node.rs

```diff
@@ -0,0 +1,574 @@
+            let child = Rc::new(RefCell::new(Node::Unresolved(child)));
+            // Path = 42 in binary.
+            let path = bitvec![Msb0, u8; 1, 0, 1, 0, 1, 0];
+
+            let mut uut = EdgeNode {
+                hash: None,
+                height: 0,
+                path,
+                child,
+            };
+
+            uut.calculate_hash();
+
+            assert_eq!(uut.hash, Some(expected));
+        }
+
+        mod path_matches {
+            use super::*;
+
+            #[test]
+            fn full() {
+                let key = StarkHash::from_hex_str("123456789abcdef").unwrap();
+                let child = Rc::new(RefCell::new(Node::Leaf(
+                    StarkHash::from_hex_str("abc").unwrap(),
+                )));
+
+                let uut = EdgeNode {
+                    hash: None,
+                    height: 0,
+                    path: key.view_bits().to_bitvec(),
+                    child,
+                };
+
+                assert!(uut.path_matches(key));
+            }
+
+            #[test]
+            fn prefix() {
+                let key = StarkHash::from_hex_str("123456789abcdef").unwrap();
+                let child = Rc::new(RefCell::new(Node::Leaf(
+                    StarkHash::from_hex_str("abc").unwrap(),
+                )));
+
+                let path = key.view_bits()[..45].to_bitvec();
+
+                let uut = EdgeNode {
+                    hash: None,
+                    height: 0,
+                    path,
+                    child,
+                };
+
+                assert!(uut.path_matches(key));
+            }
+
+            #[test]
+            fn suffix() {
+                let key = StarkHash::from_hex_str("123456789abcdef").unwrap();
+                let child = Rc::new(RefCell::new(Node::Leaf(
+                    StarkHash::from_hex_str("abc").unwrap(),
+                )));
+
+                let path = key.view_bits()[50..].to_bitvec();
+
+                let uut = EdgeNode {
+                    hash: None,
+                    height: 50,
+                    path,
+                    child,
+                };
+
+                assert!(uut.path_matches(key));
+            }
+
+            #[test]
+            fn middle_slice() {
+                let key = StarkHash::from_hex_str("123456789abcdef").unwrap();
+                let child = Rc::new(RefCell::new(Node::Leaf(
+                    StarkHash::from_hex_str("abc").unwrap(),
+                )));
+
+                let path = key.view_bits()[230..235].to_bitvec();
+
+                let uut = EdgeNode {
+                    hash: None,
+                    height: 230,
+                    path,
+                    child,
+                };
+
+                assert!(uut.path_matches(key));
+            }
+        }
+
+        mod merge_edge_child {
+            use super::*;
+
+            #[test]
+            fn edge_child() {
```

Comment:
Shouldn't path for `edge_child` and `uut` differ from each other (ie. not everlap), so that the test
case shows a more "natural" scenario, ie:
uut => height: 0, path_len: N
edge_child => height: N, path_len: M,
N + M <= 251

Reply:
`child_edge.path` bitvec is merged into `uut.path`, so you get the following result:
```rust
uut.merge_child_edge();
assert_eq!(uut.child, grandchild);
assert_eq!(uut.path.len(), 502);
```
I mean the test proves that the merging works, but when you measure the height it kinda looks weird :shrug: . I was thinking something like:

```rust
let edge_child = Rc::new(RefCell::new(Node::Edge(EdgeNode {
    hash: None,
    height: 189,
    path: path.as_bitslice()[189..].to_bitvec(),
    child: grandchild.clone(),
})));

let mut uut = EdgeNode {
    hash: None,
    height: 0,
    path: path.as_bitslice()[..189].to_bitvec(),
    child: edge_child,
};

uut.merge_child_edge();
assert_eq!(uut.child, grandchild);
assert_eq!(uut.path.len(), 251);
```